### PR TITLE
Second pass Ex. 21 model improvements

### PR DESCRIPTION
**Pull request by katie-lamb**
# Overview

Closes #XXXX.

What problem does this address?

* Removed paragraph layout docs from validation data
* Handled subsidiaries that spanned multiple rows and were grouped into one entity
* Removes duplicate rows from extracted tables

# Testing

How did you make sure this worked? How can a reviewer verify this?

# To-do list

```[tasklist]
- [ ] add other TODO items here if necessary! questions that need to answered, decisions that need to be made, tests that need to be run, etc.
- [ ] Update relevant documentation - like comments, docstrings, README, release notes, etc.
- [ ] Review the PR yourself and call out any questions or issues you have
```

### CODE REVIEW

File: src/mozilla_sec_eia/models/sec10k/ex_21/inference.py [hunk 1]
```diff
@@ -275,21 +295,31 @@ def extract_filings(
...
+        # hidden_states = []
         all_output_df = Ex21CompanyOwnership.example(size=0)
         extraction_metadata = Sec10kExtractionMetadata.example(size=0)
-        for logit, pred, output_df in pipe(_get_data(dataset)):
+        for output_dict in pipe(_get_data(dataset)):
             # TODO: logits and predictions are useful for debugging, do something with them?
-            logits.append(logit)
-            predictions.append(pred)
+            logits.append(output_dict["logits"])
+            predictions.append(output_dict["predictions"])
```
Comment (katie-lamb) [inline src/mozilla_sec_eia/models/sec10k/ex_21/inference.py:304]: I'm not actually using `logits` and `predictions` in the model (yet) but it's useful for debugging.

File: src/mozilla_sec_eia/models/sec10k/ex_21/inference.py [hunk 2]
```diff
@@ -275,21 +295,31 @@ def extract_filings(
 
         logits = []
         predictions = []
+        # hidden_states = []
```
Comment (katie-lamb) [inline src/mozilla_sec_eia/models/sec10k/ex_21/inference.py:298]: I haven't worked out the logic for using the hidden states (from the last layer of the model) yet, but I kept the draft of this commented out in this PR.
